RFCs/FS-1152-For-loop-with-accumulation.md

```diff
@@ -0,0 +1,272 @@
+# F# RFC FS-1152 - For loop with accumulation
+
+The design suggestion ["for-with" syntactic sugar for folds](https://github.com/fsharp/fslang-suggestions/issues/1362) has not yet been marked "approved in principle".
+
+This RFC covers the detailed proposal for this suggestion.
+
+- [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/1362)
+- [ ] Approved in principle
+- [ ] [Implementation](https://github.com/dotnet/fsharp/pull/FILL-ME-IN)
+- [x] [Discussion](https://github.com/fsharp/fslang-design/discussions/808)
+
+# Summary
+
+A new syntax for `for`-loops that allow accumulation is introduced.
+```
+for <pat> = <expr> with <pat> in <expr> do
+    <expr>
+```
+The value of the loop body is used to update the accumulator which is returned in the end.
+
```

Comment:
`for <pat> in <expr> with <pat> in <expr> do` runs the risk of being a pretty long line, we should define nice ways to break it up into multiple lines. I propose allowing `with` to be aligned to the left with `for`:

```fsharp
for <pat> = <expr>
with <pat> in <expr> do
    <expr>
```

Reply:
This should be in a different suggestion - see https://github.com/dotnet/fsharp/issues/18754#issuecomment-3101793041